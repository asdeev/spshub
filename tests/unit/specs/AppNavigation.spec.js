import { createLocalVue, shallowMount } from '@vue/test-utils';
import { expect } from 'chai';
import Router from 'vue-router';
import Vuetify from 'vuetify';
import testMap from '../../testMap';
import navigation from './fixtures/navigation';

const localVue = createLocalVue();
localVue.use(Router);
localVue.use(Vuetify);

describe('AppNavigation', () => {
  let props;

  const build = () => {
    const wrapper = shallowMount(testMap.App, {
      localVue,
      propsData: props
    });

    return {
      wrapper,
      toolbar: () => wrapper.find('.app-navigation__toolbar'),
      toolbarIcon: () => wrapper.find('app-navigation__toolbar-icon'),
      links: () => wrapper.find('app-navigation__links')
    };
  };

  beforeEach(() => {
    props = {
      navigation
    };
  });

  it('renders the component', () => {
    const { wrapper } = build();

    expect(wrapper.html()).to.matchSnapshot();
  });

  it('renders the main components', () => {
    const { toolbar, toolbarIcon, links } = build();

    expect(toolbar().exists()).to.be.true;
    expect(toolbarIcon().exists()).to.be.true;

    expect(links().exists()).to.be.true;
  });
});
