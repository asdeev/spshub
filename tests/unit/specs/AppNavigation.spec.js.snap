// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigation renders the component 1`] = `"<v-toolbar-stub scrollthreshold=\\"300\\" class=\\"app-navigation__toolbar\\"><v-toolbar-items-stub class=\\"app-navigation__pages\\"><v-btn-stub activeclass=\\"v-btn--active\\" to=\\"/home\\" tag=\\"button\\" flat=\\"true\\" type=\\"button\\"> home </v-btn-stub><v-btn-stub activeclass=\\"v-btn--active\\" to=\\"/about\\" tag=\\"button\\" flat=\\"true\\" type=\\"button\\"> about </v-btn-stub><v-btn-stub activeclass=\\"v-btn--active\\" to=\\"/events\\" tag=\\"button\\" flat=\\"true\\" type=\\"button\\"> events </v-btn-stub></v-toolbar-items-stub></v-toolbar-stub>"`;
